src/cascadia/TerminalControl/TSFInputControl.cpp

```diff
@@ -303,12 +284,7 @@ namespace winrt::Microsoft::Terminal::Control::implementation
     void TSFInputControl::_compositionCompletedHandler(CoreTextEditContext sender, const CoreTextCompositionCompletedEventArgs& /*args*/)
     {
         _inComposition = false;
-
-        // only need to do work if the current buffer has text
-        if (!_inputBuffer.empty())
-        {
-            _SendAndClearText();
-        }

```

Comment:
`_inputBuffer.empty()` doesn't really matter. What matters is whether `_activeTextStart == _inputBuffer.size()`, because `_activeTextStart` indicates the actual start of the text we currently edit.